tests/integration/test_http_failover/test.py

```diff
@@ -0,0 +1,151 @@
+    )
+    assert node1.query("SELECT count(*) FROM test_table;") == "1\n"
+    assert (
+        node2.query(
+            "SELECT count(*) FROM url('http://node1:8123/?query=SELECT+1', CSV, 'column2 UInt32', headers('Accept'='text/csv; charset=utf-8'));"
+        )
+        == "1\n"
+    )
+
+
+def test_url_invalid_hostname(started_cluster):
+    node1.query("DROP TABLE IF EXISTS test_table")
+
+    node1.query(
+        "CREATE TABLE test_table (column1 String, column2 UInt32) ENGINE = MergeTree ORDER BY column2;"
+    )
+    with pytest.raises(QueryRuntimeException):
+        node2.query(
+            "INSERT INTO FUNCTION url('http://notvalidhost:8123/?query=INSERT+INTO+test_table+FORMAT+CSV', 'CSV', 'column1 String, column2 UInt32') VALUES ('http interface', 42);"
+        )
+
+    assert node1.query("SELECT count(*) FROM test_table;") == "0\n"
+
+    with pytest.raises(QueryRuntimeException):
+        node2.query(
+            "SELECT count(*) FROM url('http://notvalidhost:8123/?query=SELECT+1', CSV, 'column2 UInt32', headers('Accept'='text/csv; charset=utf-8'));"
+        )
+
+
+def test_url_ip_change(started_cluster):
+    node1.query("DROP TABLE IF EXISTS test_table")
+    node1.query(
+        "CREATE TABLE test_table (column1 String, column2 UInt32) ENGINE = MergeTree ORDER BY column2;"
+    )
+    node2.query(
+        "INSERT INTO FUNCTION url('http://node1:8123/?query=INSERT+INTO+test_table+FORMAT+CSV', 'CSV', 'column1 String, column2 UInt32') VALUES ('first', 1);"
+    )
+    assert node1.query("SELECT count(*) FROM test_table;") == "1\n"
+
+    new_ip_host = "10.5.172.14"
+
+    started_cluster.restart_instance_with_ip_change(node1, new_ip_host)
+
+    node2.add_local_dns_entry("node1", new_ip_host)
+
+    node2.query("SYSTEM DROP DNS CACHE")
+
+    exec_query_with_retry(
+        node2,
+        "INSERT INTO FUNCTION url('http://node1:8123/?query=INSERT+INTO+test_table+FORMAT+CSV', 'CSV', 'column1 String, column2 UInt32') VALUES ('second', 2);",
+    )
+
+    node1.query("SELECT count(*) FROM test_table") == "2\n"
+    node2.query(
+        "SELECT count(*) FROM url('http://node1:8123/?query=SELECT+*+FROM+test_table+FORMAT+CSV', CSV, 'column1 String, column2 UInt32', headers('Accept'='text/csv; charset=utf-8'));"
+    ) == "2\n"
+
+
+def test_url_inaccessible_hostname(started_cluster):
+    node1.query("DROP TABLE IF EXISTS test_table")
+    node1.query(
+        "CREATE TABLE test_table (column1 String, column2 UInt32) ENGINE = MergeTree ORDER BY column2;"
+    )
+
+    new_ip_host = "10.5.172.15"
+    started_cluster.restart_instance_with_ip_change(node1, new_ip_host)
+
+    with pytest.raises(QueryRuntimeException):
+        node2.query(
+            "INSERT INTO FUNCTION url('http://node1:8123/?query=INSERT+INTO+test_table+FORMAT+CSV', 'CSV', 'column1 String, column2 UInt32') VALUES ('http interface', 42);"
+        )
```

Comment:
Yes, but is not a very useful test and it may be flapping. I make another parametrized one in PR continuation https://github.com/ClickHouse/ClickHouse/pull/52116